src/wrapper/java_vm/vm.rs

```diff
@@ -196,15 +196,36 @@ impl JavaVM {
             Err(error) => return Err(StartJvmError::LoadError(libjvm_path_string, error)),
         };
 
-        unsafe {
+        let result = unsafe {
             // Try to find the `JNI_CreateJavaVM` function in the loaded library.
             let create_fn = libjvm
                 .get(b"JNI_CreateJavaVM\0")
                 .map_err(|error| StartJvmError::LoadError(libjvm_path_string.to_owned(), error))?;
 
             // Create the JVM.
             Self::with_create_fn_ptr(args, *create_fn).map_err(StartJvmError::Create)
-        }
+        };
+
+        // Prevent libjvm from being unloaded.
+        //
+        // If libjvm is unloaded while the JVM is running, the program will crash as soon as it
+        // tries to execute any JVM code, including the many threads that the JVM automatically
+        // creates.
+        //
+        // For reasons unknown, HotSpot seems to somehow prevent itself from being unloaded, so it
+        // will work even if this `forget` call isn't here, but there's no guarantee that other JVM
+        // implementations will also prevent themselves from being unloaded.
+        //
+        // Note: `jni-rs` makes the assumption that there can only ever be a single `JavaVM`
+        // per-process and it's never possible to full destroy and unload a JVM once it's been
+        // created. Calling `DestroyJavaVM` is only expected to release some resources and
+        // leave the JVM in a poorly-defined limbo state that doesn't allow unloading.
+        // Ref: https://github.com/jni-rs/jni-rs/issues/567
```

Comment:
I took the liberty of tweaking this comment to refer to the issue that discusses making the assumption that the JVM can't be unloaded - and removed the notes about potentially storing the handle so it could hypothetically be dropped.